plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -580,10 +640,11 @@ private IndexMetadata loadIndexMetadata(String index)
 
         return doRequest(path, body -> {
             try {
-                Iterator<JsonNode> indicesIterator = OBJECT_MAPPER.readTree(body).elements();
-                if (maxNumberOfIndicesForAliasSchema > 0) {
-                    indicesIterator = Iterators.limit(indicesIterator, maxNumberOfIndicesForAliasSchema);
+                final JsonNode indices = OBJECT_MAPPER.readTree(body);
+                if (maxNumberOfIndicesForAliasSchema != 0 && indices.size() > maxNumberOfIndicesForAliasSchema) {
```

Comment:
nit: I know we have `@Min(0)` set for this property but just for clarity maybe change the first condition to only positive numbers.
```
maxNumberOfIndicesForAliasSchema > 0
```

Reply:
+1

The annotation controls the value in the config object when it's validated via normal means, but it doesn't guarantee that an improperly constructed configuration object won't be supplied, so this class should ensure the value matches its expectations.

Reply:
Then we will have a validation check as `@Min(1)`, Because setting a value equal to 0 makes no sense. Thoughts?